docs/architecture/adr/ADR047-replace-kiam-with-iam-roles-for-service-accounts.md

```diff
@@ -0,0 +1,46 @@
+# ADR047: Replace Kiam with IAM Roles for Service Accounts
+
+## Status
+
+Accepted
+
+## Context
+
+[Kiam][] was added into the GSP cluster in order to restrict IAM role access at a `Pod` and `Namespace`-level. It provides two things we care about. Kiam:
+
+1. Allows an individual Pod to assume an IAM role without granting all Pods that ability.
+1. Blocks access to the Instance Metadata Service.
+
+Kiam achieves this by running "agents" on every node which proxy access to the [Instance Metadata Service][] (aka IMS, aka `169.254.169.254`) through the Kiam "server". The Kiam server only proxies the request through to the "real" IMS iff the `Pod` and `Namespace` have the correct annotations.
```

Comment:
I think technically it wasn't just a proxy but it was providing an implementation of the IMDS that only exposed credentials appropriate to the requesting prod as determined by the pod's annotations (and those of its namespace) - it was using the IMDS itself to assume those roles in order to provide the temporary credentials to clients?

Reply:
Also I think the official acronym is IMDS rather than IMS, this also appears in other places in this document

Reply:
> I think technically it wasn't just a proxy but it was providing an implementation of the IMDS that only exposed credentials appropriate to the requesting prod as determined by the pod's annotations (and those of its namespace) - it was using the IMDS itself to assume those roles in order to provide the temporary credentials to clients?

I'm not sure i really care about the distinction unless you feel it's really necessary?